src/test/groovy/com/mateusz/jakuszko/roomforyou/service/ApartmentDbServiceSpecification.groovy

```diff
@@ -0,0 +1,142 @@
+package com.mateusz.jakuszko.roomforyou.service
+
+import com.mateusz.jakuszko.roomforyou.entity.Apartment
+import com.mateusz.jakuszko.roomforyou.entity.Reservation
+import com.mateusz.jakuszko.roomforyou.exceptions.NotFoundException
+import com.mateusz.jakuszko.roomforyou.repository.ApartmentRepository
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.test.context.SpringBootTest
+import spock.lang.Specification
+
+@SpringBootTest
+class ApartmentDbServiceSpecification extends Specification {
+
+    @Autowired
+    private ApartmentDbService service
+    @Autowired
+    ApartmentRepository repository
+
+    def "should throw an exception when there isn't such apartment"() {
+        when: "pull apartment from database which doesn't exist"
+        Optional<Apartment> apartment = service.getApartment(123)
+        apartment.orElseThrow(NotFoundException::new)
+
+        then:
+        thrown(NotFoundException)
+    }
+
+    def "should save and load apartment from db"() {
+        given:
+        Apartment apartment = createApartment()
+
+        when:
+        def savedApartment = saveAndGetApartmentFromDb(apartment)
+
+        then:
+        notThrown(NotFoundException)
+        checkIfFieldsAreCorrect(savedApartment, apartment)
+    }
+
+    def "should save apartments and return list of apartments after pull from db"() {
```

Comment:
1 test == 1 akcja. Jak dla mnie tytuł może brzmieć "Should return list of apartments"